app/main.py

```diff
@@ -1,2 +1,113 @@
+    def __init__(self) -> None:
+        self.buckets_table = [[] for _ in range(8)]
+        self.capacity = len(self.buckets_table)
+        self.resize_breakpoint = 2 / 3
+        self.old_buckets_table = None
+
+    def buckets_table_resize(self) -> None:
+        self.capacity *= 2
+        self.old_buckets_table = copy.deepcopy(self.buckets_table)
+        self.buckets_table = [[] for bucket in range(self.capacity)]
+
+        for bucket in self.old_buckets_table:
+            if len(bucket):
+                index_by_new_hash = bucket[2] % self.capacity
+                if not len(self.buckets_table[index_by_new_hash]):
+                    self.buckets_table[index_by_new_hash] = bucket
+                else:
+                    self.buckets_table[self.buckets_table.index([])] = bucket
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        try:
+            hash(key)
+        except TypeError:
+            raise TypeError(f"unhashable type: '{type(key).__name__}'")
+
+        k_v_pair = [key, value, hash(key)]
+        index = hash(key) % self.capacity
+
+        if not self.buckets_table[index]:
+            self.buckets_table[index] = k_v_pair
+        else:
+
+            while self.buckets_table[index][0] != key:
+                index = (index + 1) % self.capacity
+
+                if not self.buckets_table[index]:
+                    self.buckets_table[index] = k_v_pair
+                    break
+
+            if self.buckets_table[index][0] == key:
+                self.buckets_table[index] = k_v_pair
+
+        if ((self.capacity - self.buckets_table.count([]))
+                / self.capacity > self.resize_breakpoint):
+            self.buckets_table_resize()
+
+    def __getitem__(self, key: Hashable) -> Any:
+        for bucket in self.buckets_table:
+            if len(bucket) and key == bucket[0]:
+                return bucket[1]
+
+        raise KeyError(key)
+
+    def __len__(self) -> int:
+        return sum(1 for bucket in self.buckets_table if len(bucket))
+
+    def clear(self) -> None:
+        self.buckets_table = [[] for bucket in range(self.capacity)]
+
+    def __delitem__(self, key: Hashable) -> None:
+        for bucket in self.buckets_table:
+            if len(bucket) and bucket[0] == key:
+                bucket.clear()
+
+    def get(self, key: Hashable, default: Any = None) -> Any:
+        for bucket in self.buckets_table:
+            if len(bucket) and key == bucket[0]:
+                return bucket[1]
+
+        return default
+
+    def update(self, other_dict: "Dictionary") -> None:
+        for bucket in other_dict.buckets_table:
+            if len(bucket):
+                self.__setitem__(bucket[0], bucket[1])
+
+    def pop(self, key: Hashable, default: Any = None) -> Union[None, Any]:
+        value = None
+        for bucket in self.buckets_table:
+            if len(bucket) and bucket[0] == key:
+                value = bucket[1]
+                bucket.clear()
```

Comment:
You can reuse getitem + delitem

Reply:
Hello! Please review this changes:
1. I removed for loops wherever possible
2. To implement the __getitem__ (and other) methods, I created a collision list, which is much smaller than the hash table itself